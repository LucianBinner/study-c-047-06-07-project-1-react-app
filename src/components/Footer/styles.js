import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  text-align: center;

  a {
    color: inherit;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
  }
`;
