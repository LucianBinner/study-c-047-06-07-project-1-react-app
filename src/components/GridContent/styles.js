import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  text-align: center;
  max-width: 55rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  margin: ${theme.spacings.xhuge} 0;
`;
