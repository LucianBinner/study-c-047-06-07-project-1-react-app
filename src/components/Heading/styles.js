import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

const titleSize = {
  small: () => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: () => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: () => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: () => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = () => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  color: ${(props) =>
    props.colorDark ? theme.colors.primaryColor : theme.colors.white};

  ${({ size }) => titleSize[size]()}
  ${({ uppercase }) => titleCase(uppercase)}
`;
