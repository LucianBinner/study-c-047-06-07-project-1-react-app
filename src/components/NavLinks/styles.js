import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  display: flex;
  flex-flow: row wrap;

  @media ${theme.media.lteMedium} {
    flex-flow: column wrap;
    align-content: center;
  }
`;
