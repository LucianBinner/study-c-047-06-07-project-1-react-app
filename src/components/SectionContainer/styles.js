import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  max-width: 120rem;
  margin: 0 auto;
  padding: ${theme.spacings.large};
`;
