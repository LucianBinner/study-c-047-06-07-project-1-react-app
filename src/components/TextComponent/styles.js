import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.p`
  font-size: ${theme.font.sizes.medium};
`;
