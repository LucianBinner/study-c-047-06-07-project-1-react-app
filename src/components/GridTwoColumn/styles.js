import styled from 'styled-components';
import { theme } from '../../styles/theme';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: ${theme.spacings.large};

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
    text-align: center;
  }

  ${Title} {
    margin-bottom: ${theme.spacings.large};
  }
`;

export const TextContainer = styled.div`
  @media ${theme.media.lteMedium} {
    margin-bottom: ${theme.spacings.large};
  }
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
