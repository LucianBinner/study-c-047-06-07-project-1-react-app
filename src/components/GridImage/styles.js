import styled from 'styled-components';
import { theme } from '../../styles/theme';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
`;

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.large};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
