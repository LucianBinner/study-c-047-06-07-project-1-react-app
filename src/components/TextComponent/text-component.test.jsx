import { renderTheme } from '../../styles/render-theme';
import { TextComponent } from '.';
import { screen } from '@testing-library/react';

describe('<TextComponent />', () => {
  it('Should render', () => {
    renderTheme(<TextComponent>Content</TextComponent>);
    const component = screen.getByText('Content');
    expect(component).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<TextComponent>Content</TextComponent>);
    expect(container.firstChild).toMatchSnapshot(); // Se o snapchot atual for diferente do anterior o teste será quebrado, para isso basta digitar u para gerar um novo snapchot
    expect(container.firstChild).toMatchInlineSnapshot(`
      .c0 {
        font-size: 2.4rem;
      }

      <p
        class="c0"
      >
        Content
      </p>
    `); // O snapchot acima é gerado automaticamente
  });
});
