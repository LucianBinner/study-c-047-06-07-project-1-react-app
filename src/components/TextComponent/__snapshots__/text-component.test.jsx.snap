// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> Should match snapshot 1`] = `
.c0 {
  font-size: 2.4rem;
}

<p
  class="c0"
>
  Content
</p>
`;

exports[`<TextComponent /> Should match snapshot 2`] = `
.c0 {
  font-size: 2.4rem;
}

<p
  class="c0"
>
  Content
</p>
`;

exports[`<TextComponent /> Should render 1`] = `
.c0 {
  font-size: 2.4rem;
}

<p
  class="c0"
>
  Content
</p>
`;
